### ISSUE TITLE
Download failure after resume a task from specific situations

### ISSUE THREAD

**Issue by KrasnayaPloshchad** [TachibanaGeneralLaboratories/download-navi#230]
I think this is a regression introduced in 1.6 release. If I download an APK package from F-Droid, after a download task paused due to network error or timeout, I restart the task via ▶ button at the COMPLETED list, the download task started, but after the task finished, the package failed to install, and the installer notice the package is unknown type or corrupted.

**Comment by proninyaroslav** [comment]
1. What version of Android? 2. After the error appears (before restarting), the file is deleted on the disk? Or maybe its zero-sizes?

**Comment by Kita2345** [comment]
![Screenshot_20220215-101744~2](https://user-images.githubusercontent.com/87817690/153986397-1e3cdbad-46af-46a3-b0bb-e53a93c74ba5.png)

**Comment by KrasnayaPloshchad** [comment]
No. File was neither deleted on the disk nor zero-side.

**Comment by proninyaroslav** [comment]
Judging by the error, are you trying to move the file after download?

**Comment by Kita2345** [comment]
No, the error occurs after the download is complete

**Comment by proninyaroslav** [comment]
Please double check, maybe the checkbox is checked in the add dialog or in the Storage settings. Otherwise, I'll check for a possible bug.

**Comment by xibr** [comment]
I think the screenshot provided by @Kita2345 is not related to this issue, as the download path and the path of move after download are the same destination.

![Screenshot_20220216-120630_Download Navi](https://user-images.githubusercontent.com/20098154/154222161-57da1774-7618-444d-ba13-143355d709b0.jpg)

**Comment by proninyaroslav** [comment]
@xibr 
The download and move path can't be the same.

**Comment by proninyaroslav** [comment]
It might be worth catching this error and not showing it as it doesn't make much sense: you just can't move the file to the same point and it should be ignored. I moved this in a separate issue https://github.com/TachibanaGeneralLaboratories/download-navi/issues/232

**Comment by xibr** [comment]
> @xibr 
> The download and move path can't be the same. Therefore you are getting the error 

Exactly, if someone might have checked the checkbox move after the download complete, it will be the same destination.

**Comment by xibr** [comment]
About this issue I think we should check if the file exists when the resume button is pressed
 If not exists we must start the download from the beginning.

**Comment by proninyaroslav** [comment]
There is an option in the storage settings that allows to delete a file after an error occurs. Might be worth using it. It's hard to say why the file is corrupted after redownloading, in theory it should have been downloaded from the beginning.

**Comment by KrasnayaPloshchad** [comment]
> 1. What version of Android?

I used Samsung Galaxy S6+ Edge with Android 7, later I also accessed a Samsung Galaxy F52 5G, which installed Android 11. All of them reproduced.

**Comment by KrasnayaPloshchad** [comment]
This could be able to reproduce when I try to download packages from APKPure.

**Comment by proninyaroslav** [comment]
I can't reproduce this. Sequential steps are needed in which you get it.

**Comment by KrasnayaPloshchad** [comment]
Oh I usually set download destination to a folder other than Download folder, maybe this could affect that.

**Comment by proninyaroslav** [comment]
I use Android 10. I go to apkpure.com with a Navi browser, choosing a file, choosing a folder through a system file manager and start downloading. Download ends successfully.
